string name
string msg
---
bool ok
